lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -427,8 +446,11 @@ void FModuleOp::insertPorts(ArrayRef<std::pair<unsigned, PortInfo>> ports) {
       newNames.push_back(existingNames[oldIdx]);
       newTypes.push_back(existingTypes[oldIdx]);
       if (addAnnos)
-        newAnnos.push_back(existingAnnos[oldIdx]);
-      newSyms.push_back(existingSyms[oldIdx]);
+        newAnnos.push_back(existingAnnos.empty() ? emptyArray
+                                                 : existingAnnos[oldIdx]);
```

Comment:
maybe it would be better to just call `newAnnos.push_back(op.getAnnotationsAttrForPort(oldIDX))` here, and stop using the confusing tricky `existingAnnos` array.

Reply:
I like this idea very much! Trades a little bit of perf for much cleaner code.

Reply:
Done!